misk-crypto/src/testFixtures/kotlin/misk/crypto/testing/CryptoTestModule.kt

```diff
@@ -0,0 +1,167 @@
+package misk.crypto.testing
+
+import com.google.crypto.tink.Aead
+import com.google.crypto.tink.DeterministicAead
+import com.google.crypto.tink.HybridDecrypt
+import com.google.crypto.tink.HybridEncrypt
+import com.google.crypto.tink.KmsClient
+import com.google.crypto.tink.Mac
+import com.google.crypto.tink.PublicKeySign
+import com.google.crypto.tink.PublicKeyVerify
+import com.google.crypto.tink.StreamingAead
+import com.google.crypto.tink.aead.AeadConfig
+import com.google.crypto.tink.daead.DeterministicAeadConfig
+import com.google.crypto.tink.hybrid.HybridConfig
+import com.google.crypto.tink.mac.MacConfig
+import com.google.crypto.tink.signature.SignatureConfig
+import com.google.crypto.tink.streamingaead.StreamingAeadConfig
+import com.google.inject.TypeLiteral
+import com.google.inject.name.Names
+import misk.config.MiskConfig
+import misk.crypto.CryptoConfig
+import misk.crypto.ExternalDataKeys
+import misk.crypto.Key
+import misk.crypto.KeyAlias
+import misk.crypto.KeyResolver
+import misk.crypto.KeyType
+import misk.crypto.ServiceKeys
+import misk.crypto.internal.AeadEnvelopeProvider
+import misk.crypto.internal.DeterministicAeadProvider
+import misk.crypto.internal.DigitalSignatureSignerProvider
+import misk.crypto.internal.DigitalSignatureVerifierProvider
+import misk.crypto.internal.HybridDecryptProvider
+import misk.crypto.internal.HybridEncryptProvider
+import misk.crypto.internal.MacProvider
+import misk.crypto.internal.StreamingAeadProvider
+import misk.crypto.pgp.PgpDecrypter
+import misk.crypto.pgp.PgpEncrypter
+import misk.crypto.pgp.internal.PgpDecrypterProvider
+import misk.crypto.pgp.internal.PgpEncrypterProvider
+import misk.inject.KAbstractModule
+import org.bouncycastle.jce.provider.BouncyCastleProvider
+import java.security.Security
+
+/**
+ * This module should be used for testing purposes only.
+ * It generates random keys for each key name specified in the configuration
+ * and uses [FakeKmsClient] instead of a real KMS service.
+ *
+ * This module **will** read the exact same configuration files as the real application,
+ * but **will not** use the key material specified in the configuration.
+ * Instead, it'll generate a random keyset handle for each named key.
+ */
+class CryptoTestModule(
+  private val config: CryptoConfig? = null
+) : KAbstractModule() {
+  override fun configure() {
+    bind<KmsClient>().toInstance(FakeKmsClient())
+
+    AeadConfig.register()
```

Comment:
Off-topic: side-effects during `configure()` are bad, but that has nothing to do with this PR.

(And I can’t suggest a better alternative; this is the JDK’s fault for having global singletons for crypto things.)